lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -254,6 +260,8 @@ def configure(conf)
               cluster_name_from_kube_env(@kube_env)
             detect_cloudfunctions(attributes)
           end
+        elsif @subservice_name
```

Comment:
I think this should probably override 'detect_subservice' to make it more consistent with the way the other parameters work (i.e. 'project_id' is always used regardless of 'use_metadata_service').

Document this where the parameter is declared.